source/common/router/scoped_rds.cc

```diff
@@ -206,16 +212,19 @@ ScopedRdsConfigSubscription::removeScopes(
       }
       scope_name_by_hash_.erase(iter->second->scopeKey().hash());
       scoped_route_map_.erase(iter);
-      applyConfigUpdate([scope_name](ConfigProvider::ConfigConstSharedPtr config)
-                            -> ConfigProvider::ConfigConstSharedPtr {
-        auto* thread_local_scoped_config =
-            const_cast<ScopedConfigImpl*>(static_cast<const ScopedConfigImpl*>(config.get()));
-        thread_local_scoped_config->removeRoutingScope(scope_name);
-        return config;
-      });
+      removed_scope_names.push_back(scope_name);
       ENVOY_LOG(debug, "srds: remove scoped route '{}', version: {}", scope_name, version_info);
```

Comment:
Same as prior comment, I suggest modifying the wording to clarify that the update has not yet been applied.